src/core/map.cpp

```diff
@@ -1047,7 +1047,9 @@ void Map::addCustomLayer(const QString &id, std::unique_ptr<CustomLayerHostInter
 
     /note The source must exist prior to adding a layer.
 */
-void Map::addLayer(const QVariantMap &params, const QString &before) {
+void Map::addLayer(const QString &id, const QVariantMap &params, const QString &before) {
+    Q_UNUSED(id)
```

Comment:
Would it be useful to include the `id` in the "Unable to add layer" below?

Reply:
Done.